internal/service/dms/endpoint_data_source.go

```diff
@@ -0,0 +1,541 @@
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"service_access_role_arn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"stream_arn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"kms_key_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"mongodb_settings": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"auth_mechanism": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"auth_source": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"auth_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"docs_to_investigate": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"extract_doc_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"nesting_level": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"password": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"port": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"redis_settings": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"auth_password": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"auth_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"auth_user_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"port": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"server_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"ssl_ca_certificate_arn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"ssl_security_protocol": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"redshift_settings": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"bucket_folder": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"bucket_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"encryption_mode": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"server_side_encryption_kms_key_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"service_access_role_arn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"s3_settings": {
+				Description: "This argument is deprecated and will be removed in a future version; use aws_dms_s3_endpoint instead",
```

Comment:
We can probably remove this for the data source. Since its only responsible for reading the settings (not adding them), it can't really be replaced by this suggested resource.